aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp

```diff
@@ -271,7 +286,13 @@ namespace Aws
 
             void OpenSSLCipher::Init()
             {
-                EVP_CIPHER_CTX_init(&m_ctx);
+#if OPENSSL_VERSION_NUMBER < 0x10100003L
+                m_ctx = &_m_ctx;
+                EVP_CIPHER_CTX_init(m_ctx);
+#else
+                m_ctx = EVP_CIPHER_CTX_new();
```

Comment:
Uggh, I just tried to make this not need all the #ifdef checks until I realized the *new and *free functions where not part of the 1.0 interface. I'll be merging this. 

In the meantime the new and free needs to be moved to constructor and destructors, then I'll merge it over.